packages/appsync-emulator-serverless/__test__/schemaTest.test.js

```diff
@@ -314,4 +314,39 @@ describe('creates executable schema', () => {
 
     expect(result).toMatchObject({ data: { jsonTest: { test: 'yup' } } });
   });
+
+  it('custom scalar support', async () => {
+    async function asyncForEach(array, callback) {
+      for (let index = 0; index < array.length; index += 1) {
+        await callback(array[index], index, array);
+      }
+    }
+
+    const scalars = [
+      ['dateTest', new Date('05 October 2011').toISOString().split('T')[0]],
+      ['timeTest', new Date('05 October 2011').toISOString().split('T')[1]],
+      ['dateTimeTest', new Date('05 October 2011').toISOString()],
+      ['emailTest', 'foobar@example.com'],
+      ['phoneTest', '123-123-1234'],
+      ['urlTest', 'http://google.com'],
+    ];
+
+    await asyncForEach(scalars, async scalar => {
+      const [field, val] = scalar;
+
+      const source = `
+        query {
+          ${field}(${field}: "${val}")
+        }
+      `;
+
+      const result = await graphql({
+        schema,
+        contextValue,
+        source,
+      });
+
+      expect(result).toMatchObject({ data: { [field]: val } });
+    });
+  });
```

Comment:
To be honest I would prefer to have individual unit tests for each scalar in case we need to do more rigorous testing for a particular scalar, but this is okay for now unless you're inclined to change.

Reply:
I can do that. I was just attempting to be a bit DRY. Switching to individual tests will also eliminate the need for an async foreach.